rmw_fastrtps_shared_cpp/src/rmw_wait.cpp

```diff
@@ -114,89 +55,94 @@ __rmw_wait(
   // error.
   // - Heap is corrupt.
   // In all three cases, it's better if this crashes soon enough.
-  auto wait_set_info = static_cast<CustomWaitsetInfo *>(wait_set->data);
-  std::mutex * conditionMutex = &wait_set_info->condition_mutex;
-  std::condition_variable * conditionVariable = &wait_set_info->condition;
+  auto fastdds_wait_set = static_cast<eprosima::fastdds::dds::WaitSet *>(wait_set->data);
+  bool no_has_wait = false;
 
   if (subscriptions) {
     for (size_t i = 0; i < subscriptions->subscriber_count; ++i) {
       void * data = subscriptions->subscribers[i];
       auto custom_subscriber_info = static_cast<CustomSubscriberInfo *>(data);
-      custom_subscriber_info->listener_->attachCondition(conditionMutex, conditionVariable);
+      no_has_wait |= (0 < custom_subscriber_info->data_reader_->get_unread_count());
```

Comment:
I would use `RETCODE_OK == data_reader_->get_first_untaken_info()`, which means the reader has something to take.

Same for the rest of `get_unread_count` below

Reply:
Done in 5ff6c5e2